query/language/undefine.feature

```diff
@@ -495,1115 +713,1557 @@ Feature: TypeQL Undefine Query
       """
-      define
-      name owns name;
+      match $x owns name;
+      """
+    Given uniquely identify answer concepts
+      | x            |
+      | label:person |
+    Given transaction commits
+
+    Given connection open write transaction for database: typedb
+    Given typeql write query
+      """
+      insert $x isa person, has email "anon@typedb.com";
       """
     Given transaction commits
 
-    Given session opens transaction of type: write
+    Given connection open schema transaction for database: typedb
     When typeql schema query
       """
-      undefine
-      name owns name;
-      name sub attribute;
+      undefine owns name from person;
       """
     Then transaction commits
 
-    When session opens transaction of type: read
-    Then typeql throws exception
+    When connection open read transaction for database: typedb
+    Then typeql read query; fails with a message containing: "empty-set for some variable"
       """
-      match $x type name;
+      match $x owns name;
       """
 
 
-  Scenario: undefining the value type of an attribute throws an error
-    When typeql schema query; throws exception
+  Scenario: removing an attribute ownership errors if it is owned by existing instances
+    Given transaction closes
+    Given connection open write transaction for database: typedb
+    Given typeql write query
       """
-      undefine name value string;
+      insert $x isa person, has name "Tomas", has email "tomas@typedb.com";
       """
+    Given transaction commits
 
+    Given connection open schema transaction for database: typedb
+    Then typeql schema query; fails
+      """
+      undefine owns name from person;
+      """
 
-  Scenario: all existing instances of an attribute type must be deleted in order to undefine it
-    Given get answers of typeql read query
+
+  Scenario: undefining a key ownership errors if it is owned by existing instances
+    Given transaction closes
+    Given connection open write transaction for database: typedb
+    Given typeql write query
       """
-      match $x sub attribute;
+      insert $x isa person, has name "Daniel", has email "daniel@typedb.com";
       """
-    Given uniquely identify answer concepts
-      | x               |
-      | label:name      |
-      | label:email     |
-      | label:attribute |
-    Given connection close all sessions
-    Given connection open data session for database: typedb
-    Given session opens transaction of type: write
-    Given typeql insert
+    Given transaction commits
+
+    Given connection open schema transaction for database: typedb
+    Then typeql schema query; fails
+      """
+      undefine owns email from person;
+      """
+
+  ###########
+  # STRUCTS #
+  ###########
+
+    # TODO 3.x: Add tests for structs
+
+  #############
+  # FUNCTIONS #
+  #############
+# TODO: Write new tests for functions. Consider old Rules tests to be reimplemented if applicable
```

Comment:
Left old rules tests just in case. Will be happy to remove it completely.

Reply:
I think there are useful error cases here we can retain, so let's not delete them yet and then transform them into Function tests when we are there. In fact, @krishnangovindraj  please add to your list revisiting these undefine tests for but rewritten for Functions!